modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java

```diff
@@ -277,15 +282,75 @@ else if (Objects.equals(
 	}
 
 	private static void _validateNewValue(
-			Object existingValue, String objectFieldName, Object value)
+			String dbType, Object existingValue, String objectFieldName,
+			Object value)
 		throws PortalException {
 
-		if (!((Validator.isNull(existingValue) && Validator.isNull(value)) ||
-			  Objects.equals(existingValue, value))) {
+		if (Validator.isNull(existingValue) && Validator.isNull(value)) {
+			return;
+		}
+
+		if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_BLOB) ||
+			Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_CLOB) ||
+			Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_STRING)) {
+
+			if (Objects.equals(
+					GetterUtil.getString(value),
+					GetterUtil.getString(existingValue))) {
+
+				return;
+			}
+		}
+		else if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_BOOLEAN)) {
+			if (Objects.equals(
+					GetterUtil.getBoolean(value),
+					GetterUtil.getBoolean(existingValue))) {
 
-			throw new ObjectFieldReadOnlyException(
-				"Object field " + objectFieldName + " is read only");
+				return;
+			}
 		}
+		else if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_DATE) ||
+				 Objects.equals(
+					 dbType, ObjectFieldConstants.DB_TYPE_DATE_TIME)) {
+
+			String existingDateValue = String.valueOf(existingValue);
+
+			if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_DATE)) {
+				existingDateValue = existingDateValue.substring(0, 10);
+			}
+			else if (Objects.equals(
+						dbType, ObjectFieldConstants.DB_TYPE_DATE_TIME)) {
+
+				existingDateValue = existingDateValue.substring(0, 16);
+				existingDateValue = existingDateValue.replaceAll(
```

Comment:
Why do we need to get substring for date-time values? Also, why do we need to replace characters by space?
I'm afraid of these manipulations goes wrong.
Did you check/try other approaches used in Portal, like [DateUtil](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/DateUtil.java) or [CompareDatesFunction](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/CompareDatesFunction.java)